src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs

```diff
@@ -77,10 +77,17 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             codeWriter.Indent++;
             codeWriter.WriteLine("(methodInfo, options) =>");
             codeWriter.StartBlock();
-            codeWriter.WriteLine(@"Debug.Assert(options?.EndpointBuilder != null, ""EndpointBuilder not found."");");
-            codeWriter.WriteLine($"options.EndpointBuilder.Metadata.Add(new SourceKey{endpoint!.EmitSourceKey()});");
+            codeWriter.WriteLine("if (options is not { } rdfOptions)");
```

Comment:
I thought based on our offline conversation that the issue was with new usage accessing the `RequestDelegateFactoryOptions` directly. I see here that we are still only using the `options.EndpointBuilder` and not the options object in the code. How/where was the previous code presenting null dereference errors? The check here:

https://github.com/dotnet/aspnetcore/blob/5d962547339f747410304d5d3c35c25e43f6a428/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTestBase.cs#L71

should already be validating they don't show up in the generated code from the integration tests.

Reply:
Went back to using Debug.Assert(...) and cleaned up ? and ! usages within that local function.